web/src/pages/ourdna/OurDnaDashboard.tsx

```diff
@@ -0,0 +1,237 @@
+import React from 'react'
+import { CardGroup, Container } from 'semantic-ui-react'
+
+import { Box, Flex, Image, Grid, GridItem, useBreakpointValue } from '@chakra-ui/react'
+
+import { gql, useQuery } from '@apollo/client'
+// import { gql } from '../../__generated__/gql'
+
+import './OurDnaDashboard.css'
+
+import Tile from '../../shared/components/ourdna/Tile'
+import TableTile from '../../shared/components/ourdna/TableTile'
+import ThreeStatTile from '../../shared/components/ourdna/ThreeStatTile'
+
+import LoadingDucks from '../../shared/components/LoadingDucks/LoadingDucks'
+import MuckError from '../../shared/components/MuckError'
+import OurDonutChart from '../../shared/components/ourdna/OurDonutChart'
+import BarChart from '../../shared/components/ourdna/BarChart'
+
+const GET_OURDNA_DASHBOARD = gql(`
+  query DashboardQuery($project: String!) {
+    project(name: $project) {
+      ourdnaDashboard
+    }
+  }`)
+
+const Dashboard = () => {
+    const { loading, error, data } = useQuery(GET_OURDNA_DASHBOARD, {
+        variables: { project: 'greek-myth' },
+    })
+
+    const containerSizesRow = useBreakpointValue({
+        base: 'repeat(2, 1fr)',
+        sm: 'repeat(8, 1fr)',
+        md: 'repeat(2, 1fr)',
+        lg: 'repeat(2, 1fr)',
+    })
+    const containerSizesColumn = useBreakpointValue({
+        base: 'repeat(5, 1fr)',
+        sm: 'repeat(1, 1fr)',
+        md: 'repeat(5, 1fr)',
+        lg: 'repeat(5, 1fr)',
+    })
+    const tileSizesRow = useBreakpointValue({
+        base: 'repeat(2, 1fr)',
+        sm: 'repeat(4, 1fr)',
+        md: 'repeat(4, 1fr)',
+        lg: 'repeat(2, 1fr)',
+    })
+    const tileSizesColumn = useBreakpointValue({
+        base: 'repeat(2, 1fr)',
+        sm: 'repeat(1, 1fr)',
+        md: 'repeat(1, 1fr)',
+        lg: 'repeat(2, 1fr)',
+    })
+
+    const samplesLostAfterCollections = {}
+    data &&
+        Object.keys(data.project.ourdnaDashboard[0].samples_lost_after_collection).forEach(
+            (key) => {
+                samplesLostAfterCollections[key] =
+                    data.project.ourdnaDashboard[0].samples_lost_after_collection[key]
+                        .time_to_process_start / 3600
+            },
+            []
+        )
+
```

Comment:
I got badly burnt by this in my own side app with very large lists, it caused a huge amount of object creations and changing to a straight update (or maybe lodash key by) reduced it from minutes to seconds.

Reply:
Yeah that's a fair call when you're hitting performance problems, I'd still mutate within the reduce rather than mutating outside of it if the performance isn't good enough with the spread operator.

```ts
const samplesLostAfterCollections = Object
    .entries(data.project.ourdnaDashboard[0].samples_lost_after_collection)
    .reduce((rr: Record<string, number>, [key, sample]) => {
        rr[key] = sample.time_to_process_start / 3600;
        return rr;
    }, {})

```